src/extension.ts

```diff
@@ -502,11 +509,13 @@ class Completion {
 	async insertText(text: string, lineSeparator: string): Promise<number> {
 		text = lineSeparator === LF ? text : text.replaceAll(LF, lineSeparator);
 		const edit = new vscode.WorkspaceEdit();
+		const anchorPosition = this.document.positionAt(this.anchorOffset);
 		edit.insert(
 			this.document.uri,
-			this.document.positionAt(this.anchorOffset),
+			anchorPosition,
 			text
 		);
+		this.textEditor.setDecorations(this.completionIndicator, [new vscode.Range(anchorPosition, anchorPosition)]);
```

Comment:
Since `insertText` method is called multiple times, wouldn't it be better to call `setDecorations` in the constructor?

Reply:
`anchorOffset` at the time the constructor is called does not point at the end of completed text.